core/src/test/java/io/questdb/griffin/wal/WalWriterFuzzTest.java

```diff
@@ -0,0 +1,679 @@
+// using current execution time.
+// This improves coverate. To debug failures in CI find the line logging random seeds
+// and change line
+// Rnd rnd = TestUtils.generateRandom(LOG);
+// to
+// Rnd rnd = new Rnd(A, B);
+// where A, B are seeds in the failed run log.
+//
+// When the same timestamp is used in multiple transactions
+// the order of records when executed in parallel WAL writing is not guaranteed.
+// The creates failures in tests that assume that the order of records is preserved.
+// There are already measures to prevent invalid data generation but it still can happen.
+// In order to verify that the test is not broken we check that there are no duplicate timestamps for the record where the comparison fails.
+//
+public class WalWriterFuzzTest extends AbstractGriffinTest {
+
+    protected final WorkerPool sharedWorkerPool = new TestWorkerPool(4, metrics);
+    private int initialRowCount;
+    private int fuzzRowCount;
+    private boolean isO3;
+    private double cancelRowsProb;
+    private double notSetProb;
+    private double nullSetProb;
+    private double rollbackProb;
+    private double collAddProb;
+    private double collRemoveProb;
+    private double colRenameProb;
+    private double dataAddProb;
+    private int transactionCount;
+    private int strLen;
+    private int symbolStrLenMax;
+    private int symbolCountMax;
+    private int partitionCount;
+
+    @BeforeClass
+    public static void setUpStatic() {
+        walTxnNotificationQueueCapacity = 16;
+        AbstractGriffinTest.setUpStatic();
+    }
+
+    @Test
+    public void testWalAddRemoveCommitFuzzInOrder() throws Exception {
+        setFuzzProbabilities(0.05, 0.2, 0.1, 0.005, 0.05, 0.05, 0.05, 1.0);
+        setFuzzCounts(false, 1_000_000, 500, 20, 1000, 20, 0, 10);
+        runFuzz(TestUtils.generateRandom(LOG));
+    }
+
+    @Test
+    public void testWalAddRemoveCommitFuzzO3() throws Exception {
+        setFuzzProbabilities(0.05, 0.2, 0.1, 0.005, 0.05, 0.05, 0.05, 1.0);
+        setFuzzCounts(true, 100_000, 500, 20, 1000, 20, 100_000, 5);
+        runFuzz(TestUtils.generateRandom(LOG));
+    }
+
+    @Test
+    public void testWalMetadataChangeHeavy() throws Exception {
+        setFuzzProbabilities(0.05, 0.2, 0.1, 0.005, 0.25, 0.25, 0.25, 1.0);
+        setFuzzCounts(false, 100_000, 300, 20, 1000, 1000, 100, 5);
+        runFuzz(TestUtils.generateRandom(LOG));
+    }
+
+    @Test
+    public void testWalWriteFullRandom() throws Exception {
+        Rnd rnd = TestUtils.generateRandom(LOG);
+        int minPage = (int) Math.round(Math.log(Files.PAGE_SIZE) / Math.log(2));
+        dataAppendPageSize = 1L << (minPage + rnd.nextInt(29 - minPage)); // MAX page size 512Kb
+        LOG.info().$("dataAppendPageSize=").$(dataAppendPageSize).$();
+        fullRandomFuzz(rnd);
+    }
+
+    @Test
+    public void testWalWriteFullRandomMultipleTables() throws Exception {
```

Comment:
This test takes 5 mins to finish on my machine with the following RNG seeds: `12402979064715L, 1666258821825L`. Would be nice to change it, so that it takes a few seconds to execute.

Reply:
Yes, I'm sure. That's on Ubuntu 22.04 x86-64.

![Screenshot from 2022-10-20 14-14-00](https://user-images.githubusercontent.com/37772591/196934023-469019d5-be4a-47ea-9112-1fc1d6a354af.png)

Reply:
Ok fixed now, minimum append page size is `262144` on all the platforms